sharing/filter_single_value.py

```diff
@@ -0,0 +1,157 @@
+"""
+This module filters data by a single rule value of any datatype.
+"""
+
+from typing import Any, Tuple, List, Union  # pylint: disable=import-error
+import re
+import pandas as pd  # pylint: disable=import-error
+from pandas.core.frame import DataFrame  # pylint: disable=import-error
+
+
+def filter_single_value(
+    column: bool,
+    pat1: str,
+    pat2: str,
+    pat7: str,
+    pat8: str,
+    pat9: str,
+    pat10: str,
+    filter_val: Union[str, int, float, None],
+    rule_is_datetime: bool,
+    rule_is_numeric: bool,
+    rule_is_char: bool,
+    intermediate_filtered_data: DataFrame,
+    datetime_variables: List[Any],
+    numeric_variables: List[Any],
+    string_variables: List[Any],
+) -> Tuple[DataFrame, bool, bool, bool]:
+
+    """The function will filter dataframe based on a single rule value.
+
+    The function will check if the current rule value is of string type.
+    If so, will check if it is datetime type, numeric type of character type
+    and accordingly filter from respective types of variables in current rule.
+    The function returns the filtered dataframe.
+
+
+    Parameters:
+        column(bool): checks whether the rule direction is column or not
+        pat1(str): the datetime pattern to check with refular expression
+        pat2(str): the datetime pattern to check with refular expression
+        pat7(str): the datetime pattern to check with refular expression
+        pat8(str): the datetime pattern to check with refular expression
+        pat9(str): the datetime pattern to check with refular expression
+        pat10(str): the datetime pattern to check with refular expression
+        filter_val(str): current single rule value
+        rule_is_datetime(bool): bool variable checks rule value is datetime type
+        rule_is_numeric(bool): bool variable checks rule value is numeric type
+        rule_is_char(bool): bool variable checks rule value is string type
+        datetime_variables(list): current rule variables of datetime type
+        numeric_variables(list): current rule variables of numeric type
+        string_variables(list): current rule variables of string type
+    Returns:
+        dataframe: filtered data
+        bool: bool variable specifies if rule is datetime, numeric or character
+    """
+
+    # First check if the single ruleValue if of string type, then check if it is
+    # of datetime type, numeric type or character type
+    if isinstance(filter_val, str):
+        for datetime_var in datetime_variables:
+
+            # Check whether the ruleValue is of datetime type
+            if (
+                re.fullmatch(pat1, str(filter_val)) is not None
```

Comment:
This code is repeated i the datetime function also. We should have a function called `parseDateTimeRuleValue` that does all this.